packages/react/src/components/TimePickerSelect/TimePickerSelect.js

```diff
@@ -46,20 +51,19 @@ export default class TimePickerSelect extends Component {
     ),
 
     /**
-     * Provide a description for the twistie icon that can be read by screen readers
+     * iconDescription is deprecated in v10 and will be removed in v11.
      */
-    iconDescription: PropTypes.string.isRequired,
+    iconDescription: deprecate(
+      PropTypes.string,
+      'The `iconDescription` prop for `TimePickerSelect` is no longer needed and has ' +
+        'been deprecated. It will be removed in the next major release.'
```

Comment:
Should we recommend `ariaLabel` usage in its place?

```suggestion
      'The `iconDescription` prop for `TimePickerSelect` is no longer needed and has ' +
        'been deprecated. It will be removed in the next major release. Use `ariaLabel` instead.'
```